client/core/core.go

```diff
@@ -2049,6 +2049,123 @@ func (c *Core) RescanWallet(assetID uint32, force bool) error {
 	return nil
 }
 
+func (c *Core) removeWallet(assetID uint32) {
+	c.walletMtx.Lock()
+	defer c.walletMtx.Unlock()
+	delete(c.wallets, assetID)
+}
+
+// RecoverWallet will retrieve some recovery information from the wallet,
+// which may not be possible if the wallet is too corrupted, disconnect and
+// destroy the old wallet, create a new one, and if the recovery information
+// was retrieved from the old wallet, send this information to the new one.
+// If force is false, this will check for active orders involving this
+// asset before initiating a rescan. WARNING: It is ill-advised to initiate
+// a wallet recovery with active orders unless the wallet db is definitely
+// corrupted and even a rescan will not save it.
+//
+// DO NOT MAKE CONCURRENT CALLS TO THIS FUNCTION WITH THE SAME ASSET.
+func (c *Core) RecoverWallet(assetID uint32, appPW []byte, force bool) error {
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return newError(authErr, "RecoverWallet password error: %w", err)
+	}
+	defer crypter.Close()
+
+	if !force {
+		for _, dc := range c.dexConnections() {
+			if dc.hasActiveAssetOrders(assetID) {
+				return newError(activeOrdersErr, "active orders for %v", unbip(assetID))
+			}
+		}
+	}
+
+	oldWallet, found := c.wallet(assetID)
+	if !found {
+		return fmt.Errorf("RecoverWallet: wallet not found for %d -> %s: %w",
+			assetID, unbip(assetID), err)
+	}
+
+	_, isRecoverer := oldWallet.Wallet.(asset.Recoverer)
+	if !isRecoverer {
+		return errors.New("wallet is not a recoverer")
+	}
+	walletDef, err := walletDefinition(assetID, oldWallet.walletType)
+	if err != nil {
+		return err
+	}
+	// Unseeded wallets shouldn't implement the Recoverer interface. This
+	// is just an additional check for safety.
+	if !walletDef.Seeded {
+		return fmt.Errorf("cannot only recover a seeded wallet")
```

Comment:
This check is getting out ahead of the error you'd hit with `asset.CreateWallet` a few lines down

client/asset/driver.go:

```go
// CreateWallet creates a new wallet. Only use Create for seeded wallet types.
func CreateWallet(assetID uint32, seedParams *CreateWalletParams) error {
```

client/asset/btc/btc.go:

```go
// Create creates a new SPV wallet.
func (d *Driver) Create(params *asset.CreateWalletParams) error {
if params.Type != walletTypeSPV {
	return fmt.Errorf("SPV is the only seeded wallet type. required = %q, requested = %q", walletTypeSPV, params.Type)
}
if len(params.Seed) == 0 {
	return errors.New("wallet seed cannot be empty")
}
```

If "recovery" is modified to do something other than `Create`, I would agree it's an unnecessarily limitation, but it's presently central to the definition of recover.  And because we call `Destroy` before doing `Create`, the consequences for not checking this first are potentially severe.